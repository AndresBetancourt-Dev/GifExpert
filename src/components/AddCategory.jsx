import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    e.preventDefault();
    setInputValue(e.target.value);
    console.log("The Function handleInputChange has been invoked");
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      console.log("The value added was " + inputValue);
      setInputValue("");
    }
    console.log("The Function handleSubmit has been invoked");
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        className="input"
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
