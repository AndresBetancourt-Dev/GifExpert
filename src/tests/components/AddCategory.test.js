import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import AddCategory from "../../components/AddCategory";

describe("Test of Component : <AddCategory/>", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("AddCategory : Component Mounts Correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("AddCategory : Handle Input Change is Called", () => {
    const input = wrapper.find("input");
    const value = "Hello world";
    input.simulate("change", {
      target: {
        value,
      },
      preventDefault: () => {},
    });
  });

  test("AddCategory : Shouldn't submit", () => {
    wrapper.find("form").simulate("submit", {
      preventDefault: () => {},
    });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("AddCategory : Simulate Input Change, Submit, setCategories", () => {
    const input = wrapper.find("input");
    const value = "Hello world";
    input.simulate("change", {
      target: {
        value,
      },
      preventDefault: () => {},
    });
    wrapper.find("form").simulate("submit", {
      preventDefault: () => {},
    });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(input.prop("value")).toBe("");
  });
});
