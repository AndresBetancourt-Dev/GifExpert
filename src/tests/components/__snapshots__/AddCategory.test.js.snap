// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Component : <AddCategory/> AddCategory : Component Mounts Correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
